packages/utils/src/helpers/index.ts

```diff
@@ -126,3 +126,35 @@ export const truncateInMiddle = (
     str.substring(str.length - endCharCount, str.length)
   );
 };
+
+/**
+ * Assert that a value is of type `never`, the type with no possible values.
+ *
+ * Useful for making the compiler throw an error if you forget to
+ * check a possible case e.g. when switching on the value of an enum.
+ */
+export const assertNever = (x: never): never => {
+  throw new Error("this should never happen");
+  return x;
+}
+
+
+export type Ok<T> = { ok: true, value: T };
+export type Err<E> = { ok: false, error: E };
+
+/**
+ * Result type used to indicate success or failure.
+ * A value of type T is returned on success (Ok), and an error of type
+ * E is returned on failure (Err).
+ *
+ * Ok and Err types can be built with the helper functions ok and err.
+ */
+export type Result<T, E> = Ok<T> | Err<E>;
```

Comment:
This is cool :D
What do you think about namespacing functions to give more context?.
Something like:
```
export type Result<T, E> = Ok<T> | Err<E>;

export const Result = {
  ok: <T>(value: T): Ok<T> => {
    return { ok: true as const, value };
  },

  err: <E>(error: E): Err<E> => {
    return { ok: false as const, error };
  },
};
```

Reply:
I like this. I was looking in to namespacing before but I didn't realise you could export a type and value with the same name in TS.